Userland/Utilities/unzip.cpp

```diff
@@ -147,5 +179,20 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return IterationDecision::Continue;
     }));
 
+    if (!success) {
+        return 1;
+    }
+
+    success = TRY(zip_file->for_each_member([&](auto zip_member) {
+        if (zip_member.is_directory) {
+            if (adjust_modification_time(zip_member) < 0) {
+                warnln("Failed setting modification time for directory {}", zip_member.name);
+                return IterationDecision::Break;
+            }
+        }
+
+        return IterationDecision::Continue;
+    }));
```

Comment:
Why are we iterating the whole ZIP a second time?

Reply:
Regarding this and the next two comments: If I set the modification date of a directory in `unpack_zip_member` it will get overwritten when files are created in this directory. Therefore, the original modification time for a directory needs to be set after all files have been unpacked. 
This was the best solution I came up with to do this.

Reply:
We should probably store a list of directories (with the necessary attributes included) while parsing the file the first time, and use that to restore the settings instead.

Reply:
Decided to make a dedicated file and host the conversion functions there since they didn't seem to fit nicely into `AK/Time` since it doesn't support conversion from absolute times to date.

If this approach is alright, I would continue to refactor the FAT code to use the conversion functions as well.

Reply:
Done